test/t/conftest.py

```diff
@@ -186,29 +186,48 @@ def partialize(
 def bash(request) -> pexpect.spawn:
 
     logfile = None
+    histfile = None
+    tmpdir = None
+    bash = None
+
     if os.environ.get("BASHCOMP_TEST_LOGFILE"):
         logfile = open(os.environ["BASHCOMP_TEST_LOGFILE"], "w")
     elif os.environ.get("CI"):
         logfile = sys.stdout
+
     testdir = os.path.abspath(
         os.path.join(os.path.dirname(__file__), os.pardir)
     )
-    env = os.environ.copy()
-    env.update(
-        dict(
-            SRCDIR=testdir,  # TODO needed at least by bashrc
-            SRCDIRABS=testdir,
-            PS1=PS1,
-            INPUTRC="%s/config/inputrc" % testdir,
-            TERM="dumb",
-            LC_COLLATE="C",  # to match Python's default locale unaware sort
-            HISTFILE="/dev/null",  # to leave user's history file alone
-        )
+
+    # Create an empty temporary file for HISTFILE.
+    #
+    # To prevent the tested Bash processes from writing to the user's
+    # history file or any other files, we will unset the variable
+    # "HISTFILE" in "test/config/bashrc".  However, the history is
+    # loaded before the first prompt is shown and "test/config/bashrc"
+    # is loaded.  We create an empty history file using tempfile for a
+    # transient value of HISTFILE.
+    histfile = tempfile.NamedTemporaryFile(
+        prefix="bash-completion-test_", delete=False
     )
 
-    tmpdir = None
-    bash = None
```

Comment:
I don't think we should remove these -- if an error is raised before their definitions within the `try` block, the `finally` block will fail for the undefined one.

Reply:
Thank you. These lines are moved to the beginning of the function. I just thought it is more symmetric to put together these four resources (`logfile`, `histfile`, `tmpdir`, and `bash`) that will be cleaned up in the `finally` block.

Reply:
Ah, missed that if that's how it was done already at the time I took a look at this.